File: python/utils/payment_gateway.py

class PaymentGateway:
    def __init__(self, gateway_name, account_number):
        self.gateway_name = gateway_name
        self.account_number = account_number

    def pay(self, amount):
        print(f"Paid {amount} via Gateway ({self.gateway_name}): Account Number - {self.account_number}")

File: java/src/main/java/org/nitk/hackathon/utils/PaymentGateway.java
package org.nitk.hackathon.utils;

public class PaymentGateway implements Payment {
    private string gatewayName;
    private string accountNumber;
    public PaymentGateway(string gatewayName, string accountNumber) {
        this.gatewayName = gatewayName;
        this.accountNumber = accountNumber;
    }

    @Override
    public Object copy() {
        return new PaymentGateway(this.gatewayName, this.accountNumber);
    }
    
    @Override
    public void pay(double amount) {
        //logic
    }
}